Calibration/IsolatedParticles/plugins/IsolatedTracksNxN.cc

```diff
@@ -2760,18 +2761,24 @@ void IsolatedTracksNxN::printTrack(const reco::Track *pTrack) {
   if (printTrkHitPattern_) {
     const reco::HitPattern &p = pTrack->hitPattern();
 
-    edm::LogVerbatim("IsoTrack") << "default ";
+    std::ostringstream st1;
+    st1 << "default ";
     for (int i = 0; i < p.numberOfAllHits(reco::HitPattern::TRACK_HITS); i++) {
       p.printHitPattern(reco::HitPattern::TRACK_HITS, i, std::cout);
     }
-    edm::LogVerbatim("IsoTrack") << "trackerMissingInnerHits() ";
+    edm::LogVerbatim("IsoTrack") << st1.str();
+    std::ostringstream st2;
+    st2 << "trackerMissingInnerHits() ";
     for (int i = 0; i < p.numberOfAllHits(reco::HitPattern::MISSING_INNER_HITS); i++) {
-      p.printHitPattern(reco::HitPattern::MISSING_INNER_HITS, i, std::cout);
+      p.printHitPattern(reco::HitPattern::MISSING_INNER_HITS, i, st2);
```

Comment:
wouldnt this be better just to rewrite the function `printHitPattern` so it can take the `LogPrint` as an input?

Reply:
Ok, let me tag @mmusich to see what he thinks about this.

Reply:
I think the way is done here is correct, also I don't see benefits in changing `printHitPattern` (which is not Tracker-specific code, by the way, but a method of any track), as doing it the way you propose would loose some generality.